### ISSUE TITLE
Missing default focus when navigating to a page with no SemanticsNode that sets namesRoute:true

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#63685]
b/164111098

## Steps to Reproduce
```dart
import 'package:flutter/material.dart';
 
void main() {
  runApp(MaterialApp(
    title: 'Navigation Basics',
    home: FirstRoute(),
  ));
}
 
class FirstRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Route'),
      ),
      body: RaisedButton(
        child: Text('Open route'),
        onPressed: () {
          Navigator.push(
            context,
            MaterialPageRoute(builder: (context) => SecondRoute()),
          );
        },
      ),
    );
  }
}
 
class SecondRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      body: Center(
        child: RaisedButton(
          onPressed: () {
            Navigator.pop(context);
          },
          child: Text('Go back!'),
        ),
      ),
    );
  }
}
```
1. turn on talk back
2. click and open the secondRoute
3. focus is lost

## Logs

```
```

```
```
